src/handlers/walletReadyEvents.ts

```diff
@@ -1,26 +1,24 @@
-import { triggerOnSwipeLayout } from '../navigation/onNavigationStateChange';
-import { getKeychainIntegrityState } from './localstorage/globalSettings';
 import { runLocalCampaignChecks } from '@/components/remote-promo-sheet/localCampaignChecks';
 import { EthereumAddress } from '@/entities';
+import { IS_TEST } from '@/env';
+import { featureUnlockChecks } from '@/featuresToUnlock';
 import WalletBackupStepTypes from '@/helpers/walletBackupStepTypes';
+import walletBackupTypes from '@/helpers/walletBackupTypes';
 import WalletTypes from '@/helpers/walletTypes';
-import { featureUnlockChecks } from '@/featuresToUnlock';
-import { AllRainbowWallets, RainbowAccount } from '@/model/wallet';
+import { logger } from '@/logger';
+import { RainbowAccount } from '@/model/wallet';
 import { Navigation } from '@/navigation';
-
-import store from '@/redux/store';
-import { checkKeychainIntegrity } from '@/redux/wallets';
 import Routes from '@/navigation/routesNames';
-import { logger } from '@/logger';
-import { IS_TEST } from '@/env';
-import { backupsStore, CloudBackupState, LoadingStates, oneWeekInMs } from '@/state/backups/backups';
-import walletBackupTypes from '@/helpers/walletBackupTypes';
+import { checkKeychainIntegrity, getSelectedWallet, getWallets } from '@/state/wallets/walletsStore';
+import { backupsStore, LoadingStates, oneWeekInMs } from '@/state/backups/backups';
 import { useNavigationStore } from '@/state/navigation/navigationStore';
+import { triggerOnSwipeLayout } from '../navigation/onNavigationStateChange';
+import { getKeychainIntegrityState } from './localstorage/globalSettings';
 
 export const runKeychainIntegrityChecks = async () => {
   const keychainIntegrityState = await getKeychainIntegrityState();
   if (!keychainIntegrityState) {
-    await store.dispatch(checkKeychainIntegrity());
+    checkKeychainIntegrity();
```

Comment:
slight change in behavior - `checkKeychainIntegrity` is an async function, we should still await here as before; the code calling `runKeychainIntegrityChecks` may assume behavior of `checkKeychainIntegrity` completing before proceeding